### ISSUE TITLE
compiling error in NMEAShared.h

### ISSUE THREAD

**Issue by Vrijheid13** [johnboiles/Helm-firmware#1]
I found an error in compiling the firmware.
Not sure what to do!?

Arduino: 1.8.0 (Linux), Board:"SparkFun Pro Micro, ATmega32U4 (5V, 16 MHz)"

In file included from sketch/NMEAMessage.h:5:0,
                 from /home/boekel/Arduino/Helm/Helm.ino:3:
sketch/NMEAShared.h:5:19: fatal error: cstring: No such file or directory
 #include <cstring>
                   ^
compilation terminated.
exit status 1

**Comment by johnboiles** [comment]
Are you compiling using the Makefile or from the arduino ide?

**Comment by Vrijheid13** [comment]
Hi John,
I am compiling theiugh te Arduin IDE 1.8.0
Regatta, Wouter


Verzonden vanaf mijn Samsung-apparaat

-------- Oorspronkelijk bericht --------
Van: John Boiles <notifications@github.com> 
Datum: 19-01-17  22:02  (GMT+01:00) 
Aan: johnboiles/Helm-firmware <Helm-firmware@noreply.github.com> 
Cc: Vrijheid13 <wboekel@home.nl>, Author <author@noreply.github.com> 
Onderwerp: Re: [johnboiles/Helm-firmware] compiling error in NMEAShared.h (#1) 

Are you compiling using the Makefile or from the arduino ide?

**Comment by johnboiles** [comment]
Yeah my understanding of the Arduino IDE is that it's `C++`-like but doesn't have all all the normal standard library stuff you might expect. You might be better off modifying the makefile.

**Comment by johnboiles** [comment]
After cloning the repo try a `make test` and a `make build` to see if it works out of the box.

**Comment by Vrijheid13** [comment]
Hi John, 
Thanks you for the reply.
I tried Google forum the make test and make build. But no succes.
Could you help? 
Regards,Wouter 


Verzonden vanaf mijn Samsung-apparaat

-------- Oorspronkelijk bericht --------
Van: John Boiles <notifications@github.com> 
Datum: 20-01-17  08:27  (GMT+01:00) 
Aan: johnboiles/Helm-firmware <Helm-firmware@noreply.github.com> 
Cc: Vrijheid13 <wboekel@home.nl>, Author <author@noreply.github.com> 
Onderwerp: Re: [johnboiles/Helm-firmware] compiling error in NMEAShared.h (#1) 

After cloning the repo try a make test and a make build to see if it works out of the box.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/johnboiles/Helm-firmware","title":"johnboiles/Helm-firmware","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/johnboiles/Helm-firmware"}},"updates":{"snippets":[{"icon":"PERSON","message":"@johnboiles in #1: After cloning the repo try a `make test` and a `make build` to see if it works out of the box."}],"action":{"name":"View Issue","url":"https://github.com/johnboiles/Helm-firmware/issues/1#issuecomment-273995642"}}}

**Comment by sprokkie** [comment]
@Vrijheid13 
Hi, die you manage to get it working, i have the same problem

**Comment by Vrijheid13** [comment]
@sprokkie 
Hi, sorry for the late reply. 
I tried it back then to make it work. And just now, I tried it with a fresh install of linux. But no succes yet.

@johnboiles 
The Makefile procedure for compiling to an Arduino is not completely NOOB proof. And I am more a hardware guy, I would help if someone with knowledge could transform the repo to an IDE environment

Regards,
Wouter

**Comment by Vrijheid13** [comment]
@sprokkie 
I saw at he raspberry pi forum, that you got this to work!?
Could you share how you have done this?

Regards,
W

**Comment by sprokkie** [comment]
Got it to work ?

 

Not the JohnBoiles solution.

Build something similair on an Arduino only nmea  0183-> seatalk.

 

The project is dead now, I bought 3  older nmea2k displays from simrad sold the old autohelmstuf.

The plan was to use the autohelm displays on the nmea2K network, my B&G plotter translate from nmea2k to nmea0183.

 

The new project is use an Arduino or Pi and communicate on the nmea2K to send oil temp and pressure and coolant temp to the B&G plotter

 

regards

 

Van: Vrijheid13 [mailto:notifications@github.com] 
Verzonden: maandag 16 oktober 2017 12:20
Aan: johnboiles/Helm-firmware <Helm-firmware@noreply.github.com>
CC: sprokkie <sprokkie4@hotmail.com>; Mention <mention@noreply.github.com>
Onderwerp: Re: [johnboiles/Helm-firmware] compiling error in NMEAShared.h (#1)

 

@sprokkie <https://github.com/sprokkie> 
I saw at he raspberry pi forum, that you got this to work!?
Could you share how you have done this?

Regards,
W
